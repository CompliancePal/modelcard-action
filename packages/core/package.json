{
  "name": "@compliancepal/modelcard-core",
  "version": "1.0.0",
  "description": "",
  "main": "dist/index.js",
  "scripts": {
    "start": "node dist/index.js",
    "build": "tsc",
    "dev": "tsc -w",
    "test": "jest --coverage",
    "test:watch": "jest --watch"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/CompliancePal/modelcard-action.git"
  },
  "author": "",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/CompliancePal/modelcard-action/issues"
  },
  "homepage": "https://github.com/CompliancePal/modelcard-action#readme",
  "devDependencies": {
    "@types/jest": "^28.1.1",
    "@types/js-yaml": "^4.0.5",
    "@types/node": "^18.11.2",
    "@types/node-fetch": "^2.6.2",
    "@types/nunjucks": "^3.2.1",
    "jest": "^28.1.1",
    "jest-presets": "*",
    "memfs": "^3.4.7",
    "ts-jest": "^28.0.4",
    "tsconfig": "*",
    "types": "*",
    "typescript": "^4.8.4"
  },
  "jest": {
    "preset": "jest-presets/jest/node"
  },
  "dependencies": {
    "@compliancepal/spectral-rulesets": "*",
    "@stoplight/spectral-core": "^1.12.2",
    "ast-types": "^0.14.2",
    "js-yaml": "^4.1.0",
    "markdownlint": "^0.25.1",
    "mlflow-integration": "*"
  }
}
