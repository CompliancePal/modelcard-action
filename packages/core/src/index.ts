export { main, MLflowPluginOptions } from './core';
