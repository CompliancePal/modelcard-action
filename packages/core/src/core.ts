import * as core from '@actions/core';
import { configureValidator } from './steps/configureValidator';
import { BaseModelCard, ExtendedModelCard } from 'types';
import { augmentModelCard } from './steps/mlflowIntegration';

export type MainProps = {
  absCustomRulesFilepath?: string;
  disableDefaultRules: boolean;
  modelCard: string;
};

export const main = async (opts: MainProps) => {
  const validator = await configureValidator(opts);
  core.info('Validator created');
  core.debug(JSON.stringify(validator.ruleset, null, 2));

  const modelCard = await validator.validate<BaseModelCard>(opts.modelCard);
  core.info('Model card validated');

  const augmentedModelCard = await augmentModelCard(
    modelCard as ExtendedModelCard,
  );

  return augmentedModelCard;
};
