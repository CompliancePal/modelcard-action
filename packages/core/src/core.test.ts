import { readFileSync } from 'fs';
import { join } from 'path';
import { main, MainProps } from './core';

describe('core', () => {
  it('validates model card with custom rules', async () => {
    const modelCard = readFileSync(
      join(__dirname, '__fixtures__/modelcard.yaml'),
      'utf-8',
    );

    const cfg: MainProps = {
      modelCard,
      disableDefaultRules: false,
      absCustomRulesFilepath: join(__dirname, 'steps/__fixtures__/rules.yaml'),
      plugins: [],
      logger: console,
    };

    expect(await main(cfg)).toMatchSnapshot();
  });
});
