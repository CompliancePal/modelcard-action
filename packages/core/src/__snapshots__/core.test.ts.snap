// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core validates model card with custom rules 1`] = `
Object {
  "model_details": Object {
    "documentation": "<!-- markdown description of model -->
# Face Detection

The model analyzed in this card detects one or more faces within an image or a video frame, and returns a box around each face along with the location of the faces' major landmarks. The model's goal is exclusively to identify the existence and location of faces in an image. It does not attempt to discover identities or demographics.
",
    "name": "Facial Detection",
    "version": Object {
      "date": "2022-06-01",
      "name": "0.0.0",
    },
  },
  "model_parameters": Object {
    "data": Array [
      Object {
        "description": "# Descr

No further description provided
",
        "name": "training",
      },
      Object {
        "description": "# Descr

No further description provided
",
        "name": "testing",
      },
    ],
    "performance_metrics": Array [
      Object {
        "raw": "https://modelcards.withgoogle.com/assets/files/Face_Detection_Model-Card-all_chart_data.tsv",
      },
    ],
  },
}
`;
